src/Microsoft.Framework.PackageManager/DependencyAnalyzer/DependencyFinder.cs

```diff
@@ -0,0 +1,218 @@
+                if (!results.Add(current.Name))
+                {
+                    continue;
+                }
+
+                var description = libLookup[current].FirstOrDefault();
+                if (description != null)
+                {
+                    foreach (var each in description.Dependencies.Select(dep => dep.Library))
+                    {
+                        stack.Push(each);
+                    }
+                }
+                else
+                {
+                    onUnresovled(current.Name);
+                }
+            }
+
+            return results;
+        }
+
+        private ISet<string> ResolveLocalDependency(string rootReference, Action<string> onUnresvoled)
+        {
+            var result = new HashSet<string>();
+            var stack = new Stack<string>();
+            var gacResolver = new GacDependencyResolver();
+
+            stack.Push(rootReference);
+
+            while (stack.Count > 0)
+            {
+                string assemblyName = stack.Pop();
+                string assemblyFilePath = _assemblyFilePathResolver(assemblyName);
+
+                if (assemblyFilePath == null)
+                {
+                    // If the assembly file doesn't exist, try to find it in the GAC.
+                    var desc = gacResolver.GetDescription(
+                        new Library { Name = assemblyName, IsGacOrFrameworkReference = true },
+                        _framework);
+
+                    if (desc == null)
+                    {
+                        onUnresvoled(assemblyName);
+                    }
+
+                    continue;
+                }
+
+                if (!result.Add(assemblyFilePath))
+                {
+                    continue;
+                }
+
+                var assemblyInfo = new AssemblyInformation(assemblyFilePath, null);
+                foreach (var reference in assemblyInfo.GetDependencies())
+                {
+                    stack.Push(reference);
+                }
+            }
+
+            return result;
+        }
+
+        private ApplicationHostContext CreateApplicationHostContext()
+        {
+            var accessor = new CacheContextAccessor();
+            var cache = new Cache(accessor);
+
+            var hostContext = new ApplicationHostContext(
+                serviceProvider: null,
+                projectDirectory: _options.Project.ProjectDirectory,
+                packagesDirectory: null,
+                configuration: Configuration,
+                targetFramework: _framework,
+                cache: cache,
+                cacheContextAccessor: accessor,
+                namedCacheDependencyProvider: null);
+
+            hostContext.DependencyWalker.Walk(hostContext.Project.Name, hostContext.Project.Version, _framework);
+
+            return hostContext;
+        }
+
+        private Func<string, string> GetAssemblyFilePathResolver()
+        {
+            if (_framework == AspNet50)
```

Comment:
Walk through CLR and CoreCLR are different. In CoreCLR you need to take in the consideration of replacing the path to framework assemblies with the path to a latest CoreCLR package assembly.